do-core/Cargo.toml

```diff
@@ -5,3 +5,5 @@ authors = ["Samuel Ortiz <sameo@linux.intel.com>"]
 edition = "2018"
 
 [dependencies]
+strum = "0.20.0" # enum utils with a derive for FromStr
+strum_macros = "0.20.1"
```

Comment:
This brings 6 new dependent crates.
If we only need the `FromStr` trait, I would prefer that we implement it instead.

Reply:
Ok we can do that, i tried to pick the least evil between those dependencies and failing DRY

Reply:
Trying is not failing :) A lot of projects are using `strum` so it is a good option. But if we need only one struct to implement `FromStr`, then we're talking about 20 lines of codes or so, vs 6 new crates. Security and maintenance wise, it makes more sense to implement our own. As the golang proverb says: "A little copying is better than a little dependency."